### ISSUE TITLE
Tab key does not focus controls in date picker

### ISSUE THREAD

**Issue by justisb** [mui-org/material-ui-pickers#1059]
## Environment

| Tech                 | Version |
| -------------------- | ------- |
| @material-ui/pickers |  3.0.0       |
| material-ui          |   4.0.0      |
| React                |  16.8.6       |
| Browser              |  Chrome 74       |
| Peer library         |  date-fns 2.0.0-alpha.27       |  |

## Steps to reproduce

1. Open a datepicker on the demo page (https://material-ui-pickers.dev/demo/datetime-picker)
2. Press the tab key

## Expected behavior

The tab key should move between the toolbar controls and days of the month, like in version 2 of material-ui-pickers.

## Actual behavior

The tab key does not seem to focus anything.

## Live example

https://material-ui-pickers.dev/demo/datetime-picker

**Comment by justisb** [comment]
Another keyboard issue- pressing Enter on the input does not open the picker (does not open the modal or inline picker on the demo page). This is seemingly also a regression since version 2.

Optimally the picker should be able to open, select a date, select a time, and close using only the keyboard so that it is accessible to keyboard-only and screenreader users.

**Comment by dmtrKovalenko** [comment]
Ideally for accessibility is to use only keyboard mode. And even make aria-hidden on open picker button. 
We will fix not focusing, but also I suppose I will make a guide of accessibility’s

**Comment by justisb** [comment]
Thanks, I'll try using the keyboard picker and set aria-hidden on the input adornment.

It'd be great if tab could focus the hour & minute options for the time picker as well- is there anything against that or it is just low priority?

**Comment by dmtrKovalenko** [comment]
Will be fixed in nearest release

**Comment by repl-sean-heintz** [comment]
Hi @dmtrKovalenko, we have a use-case where it's not feasible to display a text input. Our application allows for user-specified formatted dates (e.g. YYYY-mm-DD, MMM dd, YYYY, etc.) to be used on every page, and currently this library does not support arbitrary masks for inputs. Additionally, we have controls where from a UX perspective it only makes sense to open a date picker instead of a text input.

Keyboard accessibility for the picker is a must-have, and text inputs don't fill every scenario.

**Comment by justisb** [comment]
In the latest release, tabbing now moves between controls, though the Enter key is not captured in the datepicker, only pressing Space. Is that expected?

**Comment by bjm904** [comment]
I would expect both Enter and Space, when focused in the input, to open the dialog.
